dask/dataframe/groupby.py

```diff
@@ -1161,10 +1165,39 @@ def _aca_agg(
         token = self._token_prefix + token
         levels = _determine_levels(self.index)
 
+        first_index = self.index if not isinstance(self.index, list) else self.index[0]
+        embarrassingly_parallel = df.known_divisions and df._contains_index_name(
+            first_index if not is_index_like(first_index) else first_index.name
+        )
+
+        if embarrassingly_parallel:
+            if len(set(df.divisions[:-1])) == len(df.divisions[:-1]):
+                warnings.warn(
+                    "This 'groupby' operation would be better optimized if all the "
+                    "rows for each group were contained in a single partition. For "
+                    "example: `df.divisions = (0, 1, 3, 4)` is better than "
+                    "`df.divisions = (0, 1, 1, 1, 2, 4)`.",
```

Comment:
```suggestion
                    "This 'groupby' operation would be better optimized if all the "
                    "rows for each group were contained in a single partition. For "
                    "example: `df.divisions = (0, 1, 3, 4)` is better than "
                    "`df.divisions = (0, 1, 1, 1, 2, 4)`. If you used a `set_index` "
                    "before this `groupby`, consider explicitly passing "
                    f"`divisions={list(dict.fromkeys(df.divisions[:-1])) + divisions[-1:]}`.",
```
TIL about `dict.fromkeys`! Had never heard of this before. https://stackoverflow.com/a/7961390/17100540

Reply:
Whoa yeah that is neat!. Ok I will assign that to a var so that I don't have to duplicate it here.

Reply:
In messing around with this though I have realized that we definitely should not include data in this error message. Better to show users this neat trick :)

Reply:
Haha that's reasonable. I was hoping for copy-pasteable, but I can see how that could really blow up.